pkg/gce-cloud-provider/compute/fake-gce.go

```diff
@@ -422,6 +421,42 @@ func (cloud *FakeCloudProvider) getGlobalSnapshotURI(snapshotName string) string
 		snapshotName)
 }
 
+type FakeBlockingCloudProvider struct {
+	*FakeCloudProvider
+	snapshotToExec chan SnapshotSourceAndTarget
+	readyToExecute chan struct{}
+}
+
+type SnapshotSourceAndTarget struct {
+	Source string
+	Target string
+}
+
+// FakeBlockingCloudProvider's method adds two channels to the CreateSnapshot process in order to provide functionality to finely
+// control the order of execution of volume operations. readyToExecute signals that an CreateSnapshot operation has been called.
+// Then it cycles through the snapshotToExec channel, waiting for permission to actually create the snapshot.
+func (cloud *FakeBlockingCloudProvider) CreateSnapshot(ctx context.Context, volKey *meta.Key, snapshotName string) (*compute.Snapshot, error) {
+	cloud.readyToExecute <- struct{}{}
+	for snapshotToExec := range cloud.snapshotToExec {
+		if snapshotToExec.Source == volKey.String() && snapshotToExec.Target == snapshotName {
+			break
+		} else {
+			cloud.snapshotToExec <- snapshotToExec
+		}
+	}
+	return cloud.FakeCloudProvider.CreateSnapshot(ctx, volKey, snapshotName)
+}
+
+func FakeCreateBlockingCloudProvider(project, zone string, cloudDisks []*CloudDisk, snapshotToExec chan SnapshotSourceAndTarget, readyToExecute chan struct{}) (*FakeBlockingCloudProvider, error) {
+	fcp, err := FakeCreateCloudProvider(project, zone, cloudDisks)
+	fbcp := &FakeBlockingCloudProvider{
```

Comment:
does this really need a constructor? Since this is just filling in struct fields, why not have the caller do that?

Reply:
I like constructors, but am ok with removing this too.